feat: create project charter (#25)
loopbackio/loopback-governance: docs/project-charter.md

# LoopBack Charter

> _note: the purpose of a project charter is to provide a brief introduction_

Reviewer: i assume the text in italic (i.e. note, directions, ex) will be removed eventually and it's there now just to help with the review?

Author: Yes, these would be removed before landing the PR. I've left them there to make it easier to review.

_or other projects which are included with the work but may not be readily_
_apparent. This may help differentiate the project from other solutions in the_
_space. If you are not using this section, please indicate your intent with the_
_phrase, 'Section Intentionally Left Blank'._

ex. [K8s SIG Architecture Charter](https://github.com/kubernetes/community/blob/HEAD/sig-architecture/charter.md#in-scope)

- Maintenance of existing Node.js packages
  - Supporting latest versions of Node.js
  - Managing deprecation
- Modernisation and extension of existing Node.js packages
- Creating new Node.js packages
- Documenting and standardising interfaces for common interactions

### 1.2: Out-of-Scope (optional)

_directions: list or bullet out areas that may be seen to be related but are_
_not included in the scope of this project. This may help clarify the kind of_
_features, contributions, issues or problems the project is looking for._
_If you are not using this section, please indicate your intent with the_
_phrase, 'Section Intentionally Left Blank'._

ex. [K8s SIG Architecture Charter](https://github.com/kubernetes/community/blob/HEAD/sig-architecture/charter.md#out-of-scope)

> Section Intentionally Left Blank

Reviewer: I also looked at the one from fastify. Might be good to use it as a reference: https://github.com/fastify/fastify/blob/main/PROJECT_CHARTER.md#12-out-of-scope.

particulary:
- Support versions of Node.js at EOL (end of life) stage
- Contributions that violates the Code of Conduct

> Section Intentionally Left Blank

Reviewer: With the comment above, my suggested change: 

```suggestion
- Support versions of Node.js at EOL (end of life) stage
- Contributions that violates the [Code of Conduct](https://github.com/loopbackio/loopback-next/blob/master/docs/CODE_OF_CONDUCT.md)
```

_the project and its respective organization and repositories. If there are_
_specific rules for membership or participation in the group, list them here or_
_by reference to a governance.md document._

ex. [Node.js TSC Charter](https://github.com/nodejs/TSC/blob/HEAD/TSC-Charter.md#section-3-establishment-of-the-tsc)

TSC memberships are not time-limited. There is no maximum size of the TSC. The
size is expected to vary in order to ensure adequate coverage of important areas
of expertise, balanced with the ability to make decisions efficiently. The TSC
must have at least four members. Although there are no hard requirements, the
composition of the TSC should aim for geographic and employer diversity in the
spirit of a distributed maintenance model.

## Section 4: Roles & Responsibilities

_directions: describe the roles and responsibilities of the LoopBack Governing Body._

ex. [K8s SIG Architecture Charter](https://github.com/kubernetes/community/blob/HEAD/sig-architecture/charter.md#roles-and-organization-management)
ex. [Node.js TSC Charter](https://github.com/nodejs/TSC/blob/HEAD/TSC-Charter.md#section-4-responsibilities-of-the-tsc)

> The LoopBack TSC is responsible for:

Reviewer: Just a suggestion: 
```suggestion
The LoopBack TSC is responsible for, but not limited to:
```

_${PROJECT} Governing Body may be responsible for regarding process or project_
_operations and management. If you are not using this section, please indicate_
_your intent with the phrase, 'Section Intentionally Left Blank'._

ex. [K8s SIG Architecture Charter](https://github.com/kubernetes/community/blob/HEAD/sig-architecture/charter.md#roles-and-organization-management)
ex. [Node.js TSC Charter](https://github.com/nodejs/TSC/blob/HEAD/TSC-Charter.md#section-5-nodejs-project-operations)

The TSC and entire technical community will follow any processes as may be
specified by the OpenJS Foundation Board relating to the intake and license
compliance review of contributions, including the OpenJS Foundation IP Policy.

### Section 4.2: Decision-making, Voting, and/or Elections (optional)

_directions: describe any provisions the project makes for decision-making_
_or include the information by reference your governance.md document._
_If you are not using this section, please indicate your intent with the_
_phrase, 'Section Intentionally Left Blank'._

ex. [Node.js TSC Charter](https://github.com/nodejs/TSC/blob/HEAD/TSC-Charter.md#section-6-elections)

> Unless stated otherwise, the LoopBack TSC adopts a lazy consensus voting system,

Reviewer: For voting, I wonder if we want to have at least another "for" vote (or half of TSC agree) and no objections, before going forward.

Author: I'm open to getting at least one more explicit "for" vote. Getting half of the TSC to vote (3 votes) may be a bit difficult as we don't have any provisions to enforce participation of TSC members.

Reviewer: make sense. "at least one for vote" works for me.

where consensus is assumed when there are no outstanding objections from the
other TSC members after a stipulated period of time of no less than 72 hours.

For more important matters such as the nomination of new TSC members, a lazy
consensus for a stipulated period of no less than 1 week is used instead.

> These votes must be done through a process that's accessible to the general

Reviewer: For voting to approve/reject a contributor to maintainer, the votes are not public in the current process.

Reviewer: > Should the TSC voting process be private as well?

sounds good as it falls under the category, IMHO.

public.

### Section 4.3: Other Project Roles (optional)

_directions: describe other roles within the project, such as chairperson,_
_tech lead, collaborator, contributor, maintainer, etc. and any responsibilities or_
_rights such role confers. You can also include this information by_
_reference to your governance.md document._
_If you are not using this section, please indicate your intent with the_
_phrase, 'Section Intentionally Left Blank'._

ex. [Node.js TSC Charter](https://github.com/nodejs/TSC/blob/HEAD/TSC-Charter.md#section-8-project-roles)

> The role of a LoopBack Maintainer is given to those that the the LoopBack TSC

Reviewer: typo:
```suggestion
The role of a LoopBack Maintainer is given to those that the LoopBack TSC
```

has confidence in their general understanding of the Project's codebase, and
trusts that they are capable for seeking feedback and consensus on non-trivial
contributions. This role provides permissions to merge pull requests in the
Project's Git Repositories, and dedicated communication channels with the

> LoopBack TSC. Nomination for the role is done in accordance with

Reviewer: We have a dedicated channel for the maintainers. I assume that's what you're referring to?

```suggestion
rest of the LoopBack maintainer community. Nomination for the role is done in accordance with
```

### 2.1 Other Formal Project Relationships (optional)

_directions: describe any additional affiliations or groups that liaise with_
_the project in a formal way (such as a W3C Community Group, for example)._
_If you are not using this section, please indicate your intent with the_
_phrase, 'Section Intentionally Left Blank'._

Section Intentionally Left Blank

## Section 3: LoopBack TSC Governing Body

_directions: describe the structure of the group responsible for managing_
_the project and its respective organization and repositories. If there are_
_specific rules for membership or participation in the group, list them here or_
_by reference to a governance.md document._

ex. [Node.js TSC Charter](https://github.com/nodejs/TSC/blob/HEAD/TSC-Charter.md#section-3-establishment-of-the-tsc)

TSC memberships are not time-limited. There is no maximum size of the TSC. The
size is expected to vary in order to ensure adequate coverage of important areas
of expertise, balanced with the ability to make decisions efficiently. The TSC
must have at least four members. Although there are no hard requirements, the
composition of the TSC should aim for geographic and employer diversity in the

> spirit of a distributed maintenance model.

Reviewer: IIUC, the Growth project requirements are for the projects wanting to be "impact" stage. I wouldn't worry about it for now.
